examples/query-routing.rs

```diff
@@ -0,0 +1,139 @@
+        IpAddr::from_str("2606:4700:4700::1111").unwrap(),
+        53,
+    );
+    let udp_connect = UdpConnect::new(server_addr);
+    let tcp_connect = TcpConnect::new(server_addr);
+    let (conn, transport) =
+        dgram_stream::Connection::new(udp_connect, tcp_connect);
+    tokio::spawn(transport.run());
+    redun.add(Box::new(conn)).await.unwrap();
+    let server_addr =
+        SocketAddr::new(IpAddr::from_str("1.1.1.1").unwrap(), 53);
+    let udp_connect = UdpConnect::new(server_addr);
+    let tcp_connect = TcpConnect::new(server_addr);
+    let (conn, transport) =
+        dgram_stream::Connection::new(udp_connect, tcp_connect);
+    tokio::spawn(transport.run());
+    redun.add(Box::new(conn)).await.unwrap();
+    let conn_service = ClientTransportToSingleService::new(redun);
+    qr.add(Name::<Vec<u8>>::from_str(".").unwrap(), conn_service);
+
+    // Queries to .com go to 2001:4860:4860::8888 and 8.8.8.8.
+    let (redun, transport) = redundant::Connection::new();
+    tokio::spawn(transport.run());
+    let server_addr = SocketAddr::new(
+        IpAddr::from_str("2001:4860:4860::8888").unwrap(),
+        53,
+    );
+    let udp_connect = UdpConnect::new(server_addr);
+    let tcp_connect = TcpConnect::new(server_addr);
+    let (conn, transport) =
+        dgram_stream::Connection::new(udp_connect, tcp_connect);
+    tokio::spawn(transport.run());
+    redun.add(Box::new(conn)).await.unwrap();
+    let server_addr =
+        SocketAddr::new(IpAddr::from_str("8.8.8.8").unwrap(), 53);
+    let udp_connect = UdpConnect::new(server_addr);
+    let tcp_connect = TcpConnect::new(server_addr);
+    let (conn, transport) =
+        dgram_stream::Connection::new(udp_connect, tcp_connect);
+    tokio::spawn(transport.run());
+    redun.add(Box::new(conn)).await.unwrap();
+    let conn_service = ClientTransportToSingleService::new(redun);
+    qr.add(Name::<Vec<u8>>::from_str("com").unwrap(), conn_service);
+
+    // Queries to .nl go to 2620:fe::9 and 9.9.9.9.
+    let (redun, transport) = redundant::Connection::new();
+    tokio::spawn(transport.run());
+    let server_addr =
+        SocketAddr::new(IpAddr::from_str("2620:fe::9").unwrap(), 53);
+    let udp_connect = UdpConnect::new(server_addr);
+    let tcp_connect = TcpConnect::new(server_addr);
+    let (conn, transport) =
+        dgram_stream::Connection::new(udp_connect, tcp_connect);
+    tokio::spawn(transport.run());
+    redun.add(Box::new(conn)).await.unwrap();
+    let server_addr =
+        SocketAddr::new(IpAddr::from_str("9.9.9.9").unwrap(), 53);
+    let udp_connect = UdpConnect::new(server_addr);
+    let tcp_connect = TcpConnect::new(server_addr);
+    let (conn, transport) =
+        dgram_stream::Connection::new(udp_connect, tcp_connect);
+    tokio::spawn(transport.run());
+    redun.add(Box::new(conn)).await.unwrap();
+    let conn_service = ClientTransportToSingleService::new(redun);
+    qr.add(Name::<Vec<u8>>::from_str("nl").unwrap(), conn_service);
```

Comment:
Perhaps a convenience function (in the library or the example) would reduce boilerplate?